node/project-management-app/controllers/app-action-handler.js

```diff
@@ -0,0 +1,431 @@
+            description = await AIPService.expandDescription(description);
+          }
+          break;
+        case AIAction.GRAMMAR:
+          if (description.trim().length > 0) {
+            description = await AIPService.correctDescription(description);
+          }
+          break;
+        default:
+        // Unrecognized action.
+      }
+      // Display the (potentially unsaved) current values of the fields from the
+      // dialog, not the values from the database.
+      const userStoryData = {
+        title,
+        description,
+        status,
+        priority,
+        size,
+        assignee,
+      };
+      const userStory = new UserStory(this.userStoryId, userStoryData);
+      const user = assignee
+        ? await UserService.getUser(this.spaceName, assignee)
+        : undefined;
+      return this.buildResponse(userStory, user, /* updated= */ false);
+    } catch (e) {
+      return handleException(e);
+    }
+  }
+
+  /**
+   * Builds a response message to send back to Google Chat after updating the
+   * user story.
+   *
+   * The content of the response message depends on the type of the card that
+   * generated the action:
+   *
+   * - single-story card message: update the message's story card
+   * - story list message: update the message's story list card
+   * - single-story dialog: push a new dialog with an updated story card
+   * - story list dialog: push a new dialog with an updated story list card
+   *
+   * @param {!UserStory} userStory The updated user story.
+   * @param {?User} user The user assigned to the user story.
+   * @param {!boolean} updated Whether the user story was updated in storage.
+   * @return {Promise<Object>} A message to post back to the DM or space.
+   */
+  async buildResponse(userStory, user, updated) {
+    switch (this.cardType) {
+      case UserStoryCardType.SINGLE_MESSAGE:
+        return {
+          text: updated ? 'User story updated.' : null,
+          cardsV2: [{
+            cardId: 'userStoryCard',
+            card: new UserStoryCard(userStory, user)
+          }],
+          actionResponse: {
+            type: 'UPDATE_MESSAGE'
+          }
+        };
+      case UserStoryCardType.LIST_MESSAGE:
+        let message = await this.app.handleMyUserStories();
+        message.actionResponse = {
+          type: 'UPDATE_MESSAGE'
+        };
+        return message;
+      case UserStoryCardType.SINGLE_DIALOG:
+        return {
+          actionResponse: {
+            type: 'DIALOG',
+            dialogAction: {
+              actionStatus: {
+                statusCode: 'OK',
+                userFacingMessage: 'Saved.'
+              },
+              dialog: {
+                body: new EditUserStoryCard(userStory, user, updated)
+              }
+            }
+          }
+        };
+      case UserStoryCardType.LIST_DIALOG:
+        return this.app.handleManageUserStories();
+      default:
+        return {};
```

Comment:
This should not happen with the current implementation, but if someone changes the code so that this happens, it may be in a situation where this function is being called that is not from a function handler. So, in this case, I changed it to just return a new card message (instead of trying to update an existing card message or dialog).